backends/xnnpack/recipes/xnnpack_recipe_provider.py

```diff
@@ -144,14 +145,16 @@ def _build_int8da_intx_weight_recipe(
         else:
             weight_granularity = PerGroup(group_size=group_size)
 
-        config = Int8DynamicActivationIntxWeightConfig(
-            weight_dtype=weight_dtype,
-            weight_granularity=weight_granularity,
+        config = AOQuantizationConfig(
```

Comment:
assert weight_dtype == torch.int4 because XNNPACK doesn't support anything but 4 for groupwise. For channelwise it can do both int4 and int8.